sklearn/metrics/_plot/tests/test_roc_curve_display.py

```diff
@@ -307,9 +307,54 @@ def test_plot_roc_curve_pos_label(pyplot, response_method, constructor_name):
     else:
         display = RocCurveDisplay.from_predictions(
             y_test,
-            y_pred_not_cancer,
+            y_score_not_cancer,
             pos_label="not cancer",
         )
 
     assert display.roc_auc == pytest.approx(roc_auc_limit)
     assert trapezoid(display.tpr, display.fpr) == pytest.approx(roc_auc_limit)
+
+
+# TODO(1.8): remove
+def test_y_score_and_y_pred_deprecation():
+    y_true = np.array([0, 1, 1, 0])
+    y_score = np.array([0.1, 0.4, 0.35, 0.8])
+    y_pred = np.array([0.2, 0.3, 0.5, 0.1])
+
+    # Test that an error is raised when both y_score and y_pred are specified
+    with pytest.raises(
+        ValueError, match="`y_pred` and `y_score` cannot be both specified"
+    ):
+        RocCurveDisplay.from_predictions(y_true, y_score=y_score, y_pred=y_pred)
+
+    # Test that y_score is used when y_pred is not specified
+    display = RocCurveDisplay.from_predictions(y_true, y_score=y_score)
+    assert_allclose(display.fpr, [0, 0.5, 0.5, 1])
+    assert_allclose(display.tpr, [0, 0, 1, 1])
+
+    # Test that y_score is used when y_pred is "deprecated"
+    display = RocCurveDisplay.from_predictions(
+        y_true, y_score=y_score, y_pred="deprecated"
+    )
+    assert_allclose(display.fpr, [0, 0.5, 0.5, 1])
+    assert_allclose(display.tpr, [0, 0, 1, 1])
+
+
+def test_y_pred_deprecation_warning():
+    y_true = np.array([0, 1, 1, 0])
+    y_pred = np.array([0.1, 0.4, 0.35, 0.8])
+
+    with pytest.warns(FutureWarning, match="y_pred was deprecated in version 1.6"):
+        display = RocCurveDisplay.from_predictions(y_true, y_pred=y_pred)
```

Comment:
The fact that this test is failing with because we are missing `y_score` means that we have a regression.

@jeremiedbb Do you see a way to get around this issue.
Indeed, I was thinking that you could move the `*` such that `y_score` is a keyword only argument as well. However, then we need to define a default for it and thus deprecate it and remove it later.

At least this is the only way that I'm seeing this change possible. And if this is the case, I would be pragmactic and not go with all this hassle for both our users (when checking the documentation) and ourself (to make the maintenance).

Anyt thoughts?